translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -5410,8 +5410,20 @@ Value *SPIRVToLLVM::transValueWithoutDecoration(SPIRVValue *BV, Function *F,
   }
 
   case OpConstantNull: {
-    auto LT = transType(BV->getType());
-    return mapValue(BV, Constant::getNullValue(LT));
+    auto BVTy = BV->getType();
+    auto LT = transType(BVTy);
+    Value *NullVal;
+    // For local memory space (LDS) the NULL value is 0xFFFFFFFF, not 0x0.
+    if (BVTy->isTypePointer() && BVTy->getPointerStorageClass() == spv::StorageClassWorkgroup)
+    {
+      auto *NullValInt = ConstantInt::get(getBuilder()->getInt32Ty(), 0xFFFFFFFF, false);
```

Comment:
why does LDS use 0xFFFFFFFF as null value? is it a HW requirement or LLVM back-end's requirment?

Reply:
The NULL values for each memory space are listed here https://llvm.org/docs/AMDGPUUsage.html#memory-spaces.

Reply:
Specifically, pointer value 0 is valid in LDS (it's the start of the thread group's LDS allocation), so you cannot use 0 as the null value.

Reply:
Got it. Thanks!